kratos/spatial_containers/bounding_volume_tree.h

```diff
@@ -270,11 +270,11 @@ class _6DOP : public kDOP // AABB
     KRATOS_CLASS_POINTER_DEFINITION(_6DOP);
     _6DOP() : kDOP() {this->Initialize();}
     virtual ~_6DOP() {}
-    virtual std::size_t NumberOfDirections() const {return 3;}
+    virtual std::size_t NumberOfDirections() const override {return 3;}
 
 private:
     static const double msDirection[][3];
-    virtual const double (*Direction() const)[3];
+    virtual const double (*Direction() const override)[3];
```

Comment:
This does not compile in my machine (clang 3.9) and it still complains about the warning it tries to solve.

My guess is that there is something strange with the (***)[3]. I am not familiar with that syntax, may I ask which is the purpose of the array at the end?

Same with the other ```(*Direction() const override)[3];```

The error:
```
/home/roigcarlo/Kratos/kratos/spatial_containers/bounding_volume_tree.h:277:46: error: expected ')'
    virtual const double (*Direction() const override)[3];
                                             ^
/home/roigcarlo/Kratos/kratos/spatial_containers/bounding_volume_tree.h:277:26: note: to match this '('
    virtual const double (*Direction() const override)[3];
                         ^
/home/roigcarlo/Kratos/kratos/spatial_containers/bounding_volume_tree.h:277:28: warning: 'Direction' overrides a member function but is not
      marked 'override' [-Winconsistent-missing-override]
    virtual const double (*Direction() const override)[3];
                           ^
/home/roigcarlo/Kratos/kratos/spatial_containers/bounding_volume_tree.h:253:28: note: overridden virtual function is here
    virtual const double (*Direction() const)[3];
```